From 41153ca4f2163d2ede2d269e2a5388d7a4b73621 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 16 May 2012 11:22:28 +0200
Subject: [PATCH] systemd/switch-root.service: do not require shutdown and
 isolate to default

do not require shutdown.target and final.target

After switch-root isolate to the default target. This will load the
default target in the real root after systemd deserializes.
---
 modules.d/98systemd/switch-root.service |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/modules.d/98systemd/switch-root.service b/modules.d/98systemd/switch-root.service
index f72739e..0c41eb0 100644
--- a/modules.d/98systemd/switch-root.service
+++ b/modules.d/98systemd/switch-root.service
@@ -8,10 +8,9 @@
 [Unit]
 Description=Switch Root
 DefaultDependencies=no
-Requires=shutdown.target final.target
-After=shutdown.target final.target
 
 [Service]
 Type=oneshot
 EnvironmentFile=/etc/switch-root.conf
 ExecStart=/usr/bin/systemctl --force switch-root ${NEWROOT} ${NEWINIT}
+ExecStopPost=-/usr/bin/systemctl isolate default.target
